packages/opentelemetry/src/contextManager.ts

```diff
@@ -0,0 +1,57 @@
+import type { Context, ContextManager } from '@opentelemetry/api';
+import type { Carrier, Hub } from '@sentry/core';
+
+import { ensureHubOnCarrier, getCurrentHub, getHubFromCarrier } from './custom/hub';
+import { setHubOnContext } from './utils/contextData';
+
+function createNewHub(parent: Hub | undefined): Hub {
+  const carrier: Carrier = {};
+  ensureHubOnCarrier(carrier, parent);
+  return getHubFromCarrier(carrier);
+}
+
+// Typescript complains if we do not use `...args: any[]` for the mixin, with:
+// A mixin class must have a constructor with a single rest parameter of type 'any[]'.ts(2545)
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Wrap an OpenTelemetry ContextManager in a way that ensures the context is kept in sync with the Sentry Hub.
+ *
+ * Usage:
+ * import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+ * const SentryContextManager = wrapContextManagerClass(AsyncLocalStorageContextManager);
+ * const contextManager = new SentryContextManager();
+ */
+export function wrapContextManagerClass<ContextManagerInstance extends ContextManager>(
```

Comment:
This has been generalized so we can wrap _any_ context manager with our sentry specific behavior. Really we don't care what context manager is used, this should also work with browser-specific context managers (e.g. the is a zonejs based context manager etc).